test/functional/mempool_updatefromblock.py

```diff
@@ -38,86 +35,52 @@ def transaction_graph_test(self, size, n_tx_to_mine=None, start_input_txid='', e
 
-        if not start_input_txid:
-            start_input_txid = self.nodes[0].getblock(self.nodes[0].getblockhash(1))['tx'][0]
-
-        if not end_address:
-            end_address = self.nodes[0].getnewaddress()
-
-        first_block_hash = ''
-        tx_id = []
-        tx_size = []
+        node = self.nodes[0]
+        self.wallet = MiniWallet(node)
         self.log.info('Creating {} transactions...'.format(size))
-        for i in range(0, size):
-            self.log.debug('Preparing transaction #{}...'.format(i))
-            # Prepare inputs.
-            if i == 0:
-                inputs = [{'txid': start_input_txid, 'vout': 0}]
-                inputs_value = self.nodes[0].gettxout(start_input_txid, 0)['value']
-            else:
-                inputs = []
-                inputs_value = 0
-                for j, tx in enumerate(tx_id[0:i]):
-                    # Transaction tx[K] is a child of each of previous transactions tx[0]..tx[K-1] at their output K-1.
-                    vout = i - j - 1
-                    inputs.append({'txid': tx_id[j], 'vout': vout})
-                    inputs_value += self.nodes[0].gettxout(tx, vout)['value']
-
-            self.log.debug('inputs={}'.format(inputs))
-            self.log.debug('inputs_value={}'.format(inputs_value))
-
-            # Prepare outputs.
-            tx_count = i + 1
-            if tx_count < size:
-                # Transaction tx[K] is an ancestor of each of subsequent transactions tx[K+1]..tx[N-1].
-                n_outputs = size - tx_count
-                output_value = ((inputs_value - fee) / Decimal(n_outputs)).quantize(Decimal('0.00000001'))
-                outputs = {}
-                for _ in range(n_outputs):
-                    outputs[self.nodes[0].getnewaddress()] = output_value
-            else:
-                output_value = (inputs_value - fee).quantize(Decimal('0.00000001'))
-                outputs = {end_address: output_value}
-
-            self.log.debug('output_value={}'.format(output_value))
-            self.log.debug('outputs={}'.format(outputs))
-
-            # Create a new transaction.
-            unsigned_raw_tx = self.nodes[0].createrawtransaction(inputs, outputs)
-            signed_raw_tx = self.nodes[0].signrawtransactionwithwallet(unsigned_raw_tx)
-            tx_id.append(self.nodes[0].sendrawtransaction(signed_raw_tx['hex']))
-            tx_size.append(self.nodes[0].getrawmempool(True)[tx_id[-1]]['vsize'])
-
-            if tx_count in n_tx_to_mine:
-                # The created transactions are mined into blocks by batches.
-                self.log.info('The batch of {} transactions has been accepted into the mempool.'.format(len(self.nodes[0].getrawmempool())))
-                block_hash = self.nodes[0].generate(1)[0]
-                if not first_block_hash:
-                    first_block_hash = block_hash
-                assert_equal(len(self.nodes[0].getrawmempool()), 0)
+        self.wallet.generate(1)
```

Comment:
`scan_blocks` is slightly quicker than `generate`, so instead we can use
```suggestion
        self.wallet.scan_blocks(start=76, num=1)
```
(76 is the first block where the coinbase is mined to the miniwallet address)

Reply:
Ah, I see. I was wondering how to implement the scan_blocks() but wasn't sure. Thanks! ac8b08b